tracer/publisher.go

```diff
@@ -0,0 +1,112 @@
+package tracer
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"math/rand"
+	"net/http"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"nhooyr.io/websocket"
+	"subtrace.dev/pubsub"
+	"subtrace.dev/rpc"
+)
+
+var DefaultPublisher = newPublisher()
+
+type Publisher struct {
+	log atomic.Bool
+	ws  *websocket.Conn
+
+	ch chan *[]byte
```

Comment:
a byte slice is already a pointer to a memory range, just `chan []byte` is enough and won't copy bytes